draft-barnes-acme.md

```diff
@@ -427,6 +435,34 @@ of {{RFC7515}}.  If the value of a "nonce" header parameter is not
 valid according to this encoding, then the verifier MUST reject the
 JWS as malformed.
 
+## Key Agreement
+
+Certain elements of the protocol will require the establishment of a shared secret between the client and the server, in such a way that an entity observing the ACME protocol cannot derive the secret.  In these cases, we use a simple ECDH key exchange, based on the system used by CMS {{RFC5753}}:
+
+* Inputs:
+  * Client-generated key pair
+  * Server-generated key pair
+  * Length of the shared secret to be derived
+  * Label
+* Perform the ECDH primitive operation to obtain Z (Section 3.3.1 of {{SEC1}})
+* Select a hash algorithm according to the curve being used:
+  * For "P-256", use SHA-256
+  * For "P-384", use SHA-384
+  * For "P-512", use SHA-512
+* Derive the shared secret value using the KDF in Section 3.6.1 of {{SEC1}} using Z and the selected hash algorithm, and with the UTF-8 encoding of the label as the SharedInfo value
```

Comment:
Scaling the hash seemed to match other existing precedent, e.g., [Suite B](https://www.nsa.gov/ia/programs/suiteb_cryptography/):

> Curve P-256 for SECRET ... Curve P-384 for TOP SECRET ...
> SHA-256 for SECRET ... SHA-384 for TOP SECRET ...

Unless you have a reason *not* to scale the hash, I'm going to keep it as is.

It is 521.